### PR TITLE
Record witness access in EXTCODEHASH

### PR DESCRIPTION
**Pull request by jsign**
This PR adds access witness recording for EXTCODEHASH

### CODE REVIEW

File: core/state_processor_test.go [hunk 1]
```diff
@@ -794,3 +795,150 @@ func TestProcessVerkleContractWithEmptyCode(t *testing.T) {
...
+		0x60, 0x00, // PUSH1 0
+		0x39, // CODECOPY
+
+		0x60, 2, // PUSH1 2
+		0x60, 0x00, // PUSH1 0
+		0xF3, // RETURN
+
+		// Contract that auto-calls EXTCODEHASH
+		0x60, 42, // PUSH1 42
+	}
```
Comment (jsign) [inline core/state_processor_test.go:866]: Deploy `dummyContract` so we have some contract to use EXTCODEHASH later.

File: core/state_processor_test.go [hunk 2]
```diff
@@ -794,3 +795,150 @@ func TestProcessVerkleContractWithEmptyCode(t *testing.T) {
...
+
+		0x60, 22, // PUSH1 22
+		0x60, 0x00, // PUSH1 0
+		0xF3, // RETURN
+
+		// Contract that auto-calls EXTCODEHASH
+		0x73, // PUSH20
+		0x3a, 0x22, 0x0f, 0x35, 0x12, 0x52, 0x08, 0x9d, 0x38, 0x5b, 0x29, 0xbe, 0xca, 0x14, 0xe2, 0x7f, 0x20, 0x4c, 0x29, 0x6a,
+		0x3F, // EXTCODEHASH
+	}
```
Comment (jsign) [inline core/state_processor_test.go:882]: Deploy another contract that actually makes the `EXTCODEHASH` to the previously deployed contract.

This was required to be super clear about witness checks. Initially, I made the dummy contract "auto call" EXTCODEHASH, but then is hard to distinguish why the witness have the contract code hash (e.g: from deployment or from EXTCODEHASH exec).

File: core/state_processor_test.go [hunk 3]
```diff
@@ -794,3 +795,150 @@ func TestProcessVerkleContractWithEmptyCode(t *testing.T) {
...
+		gen.SetPoS()
+
+		if i == 0 {
+			// Create dummy contract.
+			tx, _ := types.SignTx(types.NewContractCreation(0, big.NewInt(0), 100_000, big.NewInt(875000000), dummyContract), signer, testKey)
+			gen.AddTx(tx)
+
+			// Create contract with EXTCODEHASH opcode.
+			tx, _ = types.SignTx(types.NewContractCreation(1, big.NewInt(0), 100_000, big.NewInt(875000000), extCodeHashContract), signer, testKey)
+			gen.AddTx(tx)
```
Comment (jsign) [inline core/state_processor_test.go:894]: In the first block deploy both contracts.

File: core/state_processor_test.go [hunk 4]
```diff
@@ -794,3 +795,150 @@ func TestProcessVerkleContractWithEmptyCode(t *testing.T) {
...
+		} else {
+			tx, _ := types.SignTx(types.NewTransaction(2, extCodeHashContractAddr, big.NewInt(0), 100_000, big.NewInt(875000000), nil), signer, testKey)
+			gen.AddTx(tx)
```
Comment (jsign) [inline core/state_processor_test.go:897]: In the next block do the actual execution, so if the witness shows the contract code hash address then we know its' from EXTCODEHASH execution and not contract deployments.

File: core/state_processor_test.go [hunk 5]
```diff
@@ -794,3 +795,150 @@ func TestProcessVerkleContractWithEmptyCode(t *testing.T) {
...
+	}
+
+	codeHashStateDiff := statediff[1][stateDiffIdx].SuffixDiffs[1]
+	if codeHashStateDiff.Suffix != utils.CodeKeccakLeafKey {
+		t.Fatalf("code hash invalid suffix")
+	}
+	if codeHashStateDiff.CurrentValue == nil {
+		t.Fatalf("codeHash.CurrentValue must not be empty")
+	}
+	expCodeHash := crypto.Keccak256Hash(dummyContract[12:])
```
Comment (jsign) [inline core/state_processor_test.go:937]: Avoiding magical hashes, so we know it's correct.

**Review by g11tech — APPROVED** [review]
good to see version access being added as well :+1:

**Review by gballet — APPROVED** [review]
LGTM

**Pull request merged**